src/mimefactory.rs

```diff
@@ -173,6 +176,7 @@ impl<'a> MimeFactory<'a> {
             req_mdn,
             last_added_location_id: 0,
             attach_selfavatar,
+            quoted_msg_subject: msg.quoted_message(context).await?.map(|m| m.subject),
```

Comment:
I thought about how to do this for quite a while, it's not as obvious as it looks, and I'm not sure whether I chose the best solution:
- Some lines above: https://github.com/deltachat/deltachat-core-rust/blob/cc9da03161b730c0e689309b21ab60241c715d21/src/mimefactory.rs#L146
  we select `in_reply_to` from the database instead of using `msg.in_reply_to` (`mime_in_reply_to` means the same as `in_reply_to`). `msg.quoted_message()`, however, uses `self.in_reply_to`, i.e. not the `in_reply_to` from the database but the one in memory. Which might produce inconsistencies.
   - As `set_quote()` only changes `in_reply_to` in memory, I was actually surprised that there were no problems so far:
     If `set_quote()` only changes it in memory and in the code above we read it from db, why do we ever reply to the correct message? I didn't investigate into why this works further, though.
- `quoted_message()` does two database reads, one of which is loading a whole `Message` struct. This delays sending of the message, of course.
  A previous solution was:

                let parent_subj: Option<String> = context
                    .sql
                    .query_get_value_result(
                        "SELECT subject FROM msgs WHERE rfc724_mid=?",
                        paramsv![self.in_reply_to],
                    )
                    .await?;

  It has some problems, but in generally a single query would probably be enough.

Reply:
i think, it is fine for now, as we maybe want to get quoted_msg_subject calculation out of mimefactory.rs to allow subjects set for outgoing/forwarded messages (and maybe even manually at some point :), this will change anyway in a subsequent pr.